### PR TITLE
adding cloning to views 

### PR DESCRIPTION
**Pull request by danyaljj**
- overriding clone function for views, relations and constituents 
- using the clone in the evaluators, to prevent them from modifying the contents of the views.

### CODE REVIEW

**Comment by danyaljj** [comment]
Had some CI failures. Fixed now.

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java [hunk 1]
```diff
@@ -559,4 +558,9 @@ public int count() {
         return this.constituents.size();
     }
 
+    @Override
+    public Object clone() throws CloneNotSupportedException {
+        View view = (View) super.clone();
```
Comment (bhargav) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java]: Question: Shouldn't cloning a view also clone its constituents and relations respectively? Also is the cloned View attached to the same TextAnnotation object? 

I don't understand how cloning Views and Constituents independent of their TextAnnotation would work?

Comment (danyaljj) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java]: See if it's good now?

**Comment by bhargav** [comment]
Can't set status; build failed.

**Comment by bhargav** [comment]
Can't set status; build succeeded.

File: core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java (hunk 1 above)
Comment (danyaljj) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java]: @bhargav

Comment (bhargav) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java]: Can we discuss this once tomorrow? Some concerns about cloning views and constituents.

Comment (danyaljj) [inline core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java]: 👍

**Comment by bhargav** [comment]
Can't set status; build failed.

**Pull request closed without merging**